pkg/image/containerdregistry/resolver.go

```diff
@@ -88,7 +89,13 @@ func credential(cfg *configfile.ConfigFile) func(string) (string, string, error)
 	}
 }
 
+// protects against a data race inside the docker CLI
+var configMutex sync.Mutex
+
 func loadConfig(dir string) (*configfile.ConfigFile, error) {
+	configMutex.Lock()
```

Comment:
I'm not sure how the lock here actually prevents a race condition if this function is called outside of the race context. I see uses of `docker.ConfigFile` and `docker.Credentials` here but I don't see how these benefit from a lock. 

Maybe just bumping the docker dependency to the fixed version is enough to address the issue?

Reply:
The code in the lib uses a global variable during config loading to deal with some stdout messaging. During tests loading is called multiple times and thus triggers the race detector on the global.

I think the global was introduced into the dependency somewhere between the last version we dependent on and the latest one. :shrug: 

So while bumping the docker dependency deals with the issue this MR is about, it introduces a data race in the tests. The lock solves it.

Reply:
@jeloba is there an upstream issue that documents this problem in github.com/docker/cli?

I'd prefer if we had a breadcrumb in this repo that reminds us that the lock is there due to an upstream issue and gives us a pointer to know when it's safe to remove the lock (i.e. when the upstream bug is fixed, we can bump our dep and remove the lock)

Reply:
@joelanford  yes there is https://github.com/docker/cli/pull/3410 should I reference that in the code?

Reply:
Yeah, I think my preference would be to inline that issue link as a TODO comment if we can't away from having to introduce locking mechanisms in the code.